libs/walproposer/src/walproposer.rs

```diff
@@ -327,11 +333,21 @@ mod tests {
     }
 
     impl ApiImpl for MockImpl {
+        fn get_shmem_state(&self) -> *mut crate::bindings::WalproposerShmemState {
+            self.shmem.get()
+        }
+
         fn get_current_timestamp(&self) -> i64 {
             println!("get_current_timestamp");
             0
         }
 
+        fn update_donor(&self, donor: &mut crate::bindings::Safekeeper, donor_lsn: u64) {
```

Comment:
Nit, mostly note to myself: this is redundant because MockImpl never actually uses neon walreader and donor. However it is still makes sense to have at least dummy implementation to avoid panicking on todo! when callback is called (though it doesn't seem to be really called in tests).